atomate/vasp/fireworks/core.py

```diff
@@ -831,3 +833,176 @@ def __init__(self, spec, neb_label, from_images=True,
                  PassCalcLocs(name=label)]
 
         super(NEBFW, self).__init__(tasks, spec=spec, name=label, **kwargs)
+
+
+class SurfCalcOptimizer(Firework):
+    """
+    Optimizer firework for surface property calculation and insertion. Will build
+        a Firework tailored to the type of structure and as a result, use  different
+        input sets and insert different additionals depending on the structure_type.
+    """
+
+    def __init__(self, structure, scratch_dir, k_product, vasp_cmd,
+                 structure_type, cwd, db_file=None, miller_index=None,
+                 ouc=None, scale_factor=None, vsize=None, mmi=None, mpid="--",
+                 shift=None, ssize=None, reconstruction=None,  **kwargs):
+        """
+        Initializes the Firework.
+
+        Args:
+            structure (Structure): Can be a conventional unit cell, oriented
+                unit cell or slab cell.
+            scratch_dir (str): - if specified, uses this directory as the root
+                scratch dir. Supports env_chk.
+            k_product (int): Default to 50, kpoint number * length for a & b
+                directions, also for c direction in bulk calculations.
+            db_file (str): FULL path to file containing the database credentials.
+                Supports env_chk.
+            vasp_cmd (str): Command to run vasp.
+            structure_type (str): Type of structure. Options are:
+                conventional_unit_cell, oriented_unit_cell and slab_cell
+            miller_index ([h, k, l]): Miller index of plane parallel to
```

Comment:
remember to indicate what the default value will be in the documentation if the user does not set it.